server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java

```diff
@@ -60,11 +62,22 @@
 
   private final WebService.Context context;
 
-  public WebServiceEngine(WebService[] webServices) {
+  public WebServiceEngine(WebService[] webServices, WebServiceDefinition[] webServiceDefinitions) {
     context = new WebService.Context();
     for (WebService webService : webServices) {
       webService.define(context);
     }
+    Arrays.stream(webServiceDefinitions)
+      .map(WebServiceDefinition::define)
+      .forEach(context::addController);
+  }
+
+  /**
+   * @deprecated since 6.5 use {@link #WebServiceEngine(WebService[],WebServiceDefinition[])}
+   */
+  @Deprecated
+  public WebServiceEngine(WebService[] webServices) {
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)